tc-health-client/config/config.go

```diff
@@ -80,12 +82,38 @@ func (lcfg LogCfg) InfoLog() log.LogLocation    { return log.LogLocation(lcfg.Lo
 func (lcfg LogCfg) DebugLog() log.LogLocation   { return log.LogLocation(lcfg.LogLocationDebug) }
 func (lcfg LogCfg) EventLog() log.LogLocation   { return log.LogLocation(log.LogLocationNull) } // not used
 
-func ReadCredentials(cfg *Cfg) error {
-	fn := cfg.TOCredentialFile
-	f, err := os.Open(fn)
+/**
+ * cfg - the existing config
+ * updating - when true, existing credentials may be updated from the credential file
+ */
+func ReadCredentials(cfg *Cfg, updating bool) error {
+	if cfg.TOCredentialFile == "" {
+		return nil
+	}
+
+	fn := cfg.CredentialFile
+
+	// verify that we have credentials or can read them from the credential file
+	if fn.Filename == "" {
+		if cfg.TOPass == "" || cfg.TOUser == "" || cfg.TOUrl == "" {
+			return fmt.Errorf("cannot continue, no TO credentials or TO URL have been specified, check configs")
+		} else {
+			return nil
+		}
+	}
+
+	// You should not configure a credential file and the credentials simultaneously in the health client
+	// config file.  Either use an external credential file or put the credentials in the health client
+	// config.  Precedence is given to credentials in the health client config file.
+	if !updating && (cfg.TOPass != "" && cfg.TOUser != "" && cfg.TOUrl != "") {
+		log.Warnf("Credentials are defined in the %s file, will not override them with those in the %s file", cfg.HealthClientConfigFile.Filename, cfg.CredentialFile.Filename)
+		cfg.CredentialFile.LastModifyTime = math.MaxInt64
+		return nil
+	}
 
+	f, err := os.Open(fn.Filename)
 	if err != nil {
-		return errors.New("failed to open + " + fn + " :" + err.Error())
+		return errors.New("Failed to open + " + fn.Filename + " :" + err.Error())
```

Comment:
Error strings should not be capitalized in go.

Reply:
@zrhoffman why?

Reply:
That's just one of the idioms of go. From https://github.com/golang/go/wiki/Errors:

> Error strings should not be capitalized (unless beginning with proper nouns or acronyms) or end with punctuation, since they are usually printed following other context. That is, use `fmt.Errorf("something bad")` not `fmt.Errorf("Something bad")`, so that `log.Printf("Reading %s: %v", filename, err)` formats without a spurious capital letter mid-message. This does not apply to logging, which is implicitly line-oriented and not combined inside other messages.

Reply:
@zrhoffman changes made